### PR TITLE
Make common netspeak get replaced in the chat

### PR DESCRIPTION
**Pull request by 27alaing**
## About the PR
Common internet slang gets converted into their actual meaning
EX: ong --> on god, ty --> thank you, thx --> thanks, imo --> in my opinion, etc.

i also added new phrases to the chat sanitization manager,
rofl = laugh emote
smh = unimpressed emote

## Why / Balance
this allows players to type faster, since netspeak is not allowed on the servers

also makes admins' jobs a bit easier

## Media
https://github.com/space-wizards/space-station-14/assets/45323883/061b6fa7-2b3a-402c-ba08-041a8b5f1f8d

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
None

**Changelog**
:cl:
- add: Common internet slang now gets replaced in IC chat. (example: brb --> be right back)

### CODE REVIEW

**Comment by lzk228** [comment]
it can be done without hundreds lines in fluent and prototypes 
https://github.com/space-syndicate/space-station-14/pull/733

**Comment by deltanedas** [comment]
i think this might encourage it more and let more obscure words slip through the cracks
if its faster to write pls instead of please people will probably say other things more

**Comment by UbaserB** [comment]
beautiful, beautiful work. Dont forget other slangs like iirc, smh, tf, etc.

**Comment by 27alaing** [comment]
> it can be done without hundreds lines in fluent and prototypes [space-syndicate/space-station-14#733](https://github.com/space-syndicate/space-station-14/pull/733)

thanks for this, I made it easier to add more words

**Comment by moonheart08** [comment]
im here to inform you that they gave you bad advice and you should revert to what you had before

**Comment by 27alaing** [comment]
> im here to inform you that they gave you bad advice and you should revert to what you had before

how so

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
for the dictionary, replace it with locale strings. You shouldn't have raw english text in code.

**Comment by lzk228** [comment]
> You shouldn't have raw english text in code.

SkeletonAccentSystem.cs and OwOAccentSystem.cs :trollface:

**Comment by 27alaing** [comment]
reverted it back to the way it worked before, also added some new words

**Comment by 27alaing** [comment]
fixed conflict

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 1]
```diff
@@ -687,9 +689,11 @@ private bool CanSendInGame(string message, IConsoleShell? shell = null, IPlayerS
     }
 
     // ReSharper disable once InconsistentNaming
-    private string SanitizeInGameICMessage(EntityUid source, string message, out string? emoteStr, bool capitalize = true, bool punctuate = false, bool capitalizeTheWordI = true)
+    private string SanitizeInGameICMessage(EntityUid source, string message, out string? emoteStr, bool capitalize = true, bool punctuate = false, bool capitalizeTheWordI = true, bool replaceNetSpeak = true)
     {
         var newMessage = message.Trim();
+        if (replaceNetSpeak)
+            newMessage = SanitizeMessageReplaceWords(newMessage);
         if (capitalize)
```
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/ChatSystem.cs:697]: Please don't add dummy args that are never intended to be overwritten, this class is already bloated as is.

Comment (27alaing) [reply Content.Server/Chat/Systems/ChatSystem.cs]: got it

**Comment by UbaserB** [comment]
you forgot the words “mf” and “etc” (becoming etcetera)

**Comment by 27alaing** [comment]
> you forgot the words “mf” and “etc” (becoming etcetera)

smart, added those

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 2]
```diff
@@ -737,6 +741,17 @@ private string SanitizeMessagePeriod(string message)
         return message;
     }
 
+    public string SanitizeMessageReplaceWords(string message)
+    {
+        if (string.IsNullOrEmpty(message)) return message;
+
+        var msg = message;
+
+        msg = _wordreplacement.ApplyReplacements(msg, "chatsanitize");
```
Comment (EmoGarbage404) [inline Content.Server/Chat/Systems/ChatSystem.cs:750]: make "chatsanitize" a const string with the `[ValidatePrototypeId<AccentPrototype>]` attribute on it.

Comment (27alaing) [reply Content.Server/Chat/Systems/ChatSystem.cs]: hopefully did it, is it good?

**Review by EmoGarbage404 — APPROVED** [review]

**Pull request merged**